### PR TITLE
[v0.9.4] Add PFC based pause state

### PR DESCRIPTION
**Pull request by stcheng**
Signed-off-by: Shuotian Cheng <shuche@microsoft.com>

### CODE REVIEW

File: inc/saiqueue.h [hunk 1]
```diff
@@ -70,6 +70,9 @@ typedef enum _sai_queue_attr_t
     /** Attach scheduler to Queue [sai_object_id_t]*/
     SAI_QUEUE_ATTR_SCHEDULER_PROFILE_ID = 0x00000003,
 
+    /** PFC based pause state [booldata] */
```
Comment (lguohan) [inline inc/saiqueue.h]: should not restricted to pfc, just queue pause state.
default false. (true: means queue in pause state).

Comment (stcheng) [inline inc/saiqueue.h]: updated

**Comment by AshokDaparthi** [comment]
@lguohan - is this READ_ONLY Attribute to know the pause state on queue or to enable honoring on queue ?

**Comment by tushar-ty** [comment]
@AshokDaparthi, yes this is RO for knowing the current state.

**Comment by stcheng** [comment]
@tushar-ty I will update the PR to indicate that this is RO. But if it is RO, why is there a default value? Or does it need to have a default value? What will trigger this attribute to change?

**Comment by tushar-ty** [comment]
Yes, the default value field should be removed. I did not remember if it wasn't required for RO attribs. This attrib represents the queues internal h/w state and is updated upon receiving PFC frames.

**Comment by stcheng** [comment]
@tushar-ty So here if the value is true, it means that the queue is paused?

**Comment by stcheng** [comment]
@tushar-ty Could you also take a look at the changes on master?
https://github.com/opencomputeproject/SAI/pull/405

File: inc/saiqueue.h [hunk 2]
```diff
@@ -70,6 +70,10 @@ typedef enum _sai_queue_attr_t
     /** Attach scheduler to Queue [sai_object_id_t]*/
     SAI_QUEUE_ATTR_SCHEDULER_PROFILE_ID = 0x00000003,
 
+    /** Queue pause state [bool]
+      * (READ_ONLY default is FALSE) */
```
Comment (lguohan) [inline inc/saiqueue.h]: if it is read only, then there is no default value.

**Comment by lguohan** [comment]
agree with tushar, there should be no default value for RO attribute.

File: inc/saiqueue.h (hunk 2 above)
Comment (stcheng) [inline inc/saiqueue.h]: updated

**Comment by stcheng** [comment]
🏓

**Comment by stcheng** [comment]
👆

**Pull request merged**